osu.Game/Screens/Menu/MainMenuButton.cs

```diff
@@ -48,15 +49,22 @@ public partial class MainMenuButton : BeatSyncedContainer, IStateful<ButtonState
         public ButtonSystemState VisibleState = ButtonSystemState.TopLevel;
 
         private readonly Action clickAction;
+        private readonly Action<HoverEvent> hoverAction;
         private Sample sampleClick;
         private Sample sampleHover;
 
+        // for tests
+        public bool LooksHovered { get; private set; }
+
         public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => box.ReceivePositionalInputAt(screenSpacePos);
 
-        public MainMenuButton(LocalisableString text, string sampleName, IconUsage symbol, Color4 colour, Action clickAction = null, float extraWidth = 0, Key triggerKey = Key.Unknown)
+        public MainMenuButton(ButtonSystem buttonSystem, LocalisableString text, string sampleName, IconUsage symbol, Color4 colour, Action clickAction = null, float extraWidth = 0,
```

Comment:
I really don't like that the buttons receive the entire `ButtonSystem` and call check methods on it. The `ConfirmHover()` thingy can be done by having each `MainMenuButton` expose a `HandleHover` bindable or something, and binding it to a single source of truth that would sit in `ButtonSystem`.

Reply:
since i introduced the hover action anyway, I made it now trigger for hover and unhover with position and target, which I think cleaned this up a lot and also made ButtonSystem nicer to read. Thanks for the suggestions.